### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by daaain**
# Integration Tests & Snapshot Testing

This PR adds comprehensive integration testing using real Claude Code project logs and introduces HTML snapshot testing with Syrupy.

## Integration Tests

Adds a new test_integration_realistic.py module that uses actual project log data from ~/.claude/projects/ (anonymised and committed to the test fixtures). These tests exercise the full processing pipeline with realistic data structures including:

- Multi-session projects with agent sidechains
- Various message types (tool use, thinking blocks, images, etc.)
- Cross-session summary matching
- The complete HTML rendering pipeline

The test fixtures include representative samples from several real projects, ensuring the renderer handles edge cases found in actual usage.

## Snapshot Testing

Introduces HTML snapshot testing using https://github.com/syrupy-project/syrupy to detect unintended changes to HTML output. This catches regressions in the renderer without manually checking every output format.

- Custom serialiser for readable snapshot diffs
- Covers key rendering scenarios from the integration test data
- Snapshots update via pytest --snapshot-update

## Other Changes

- Updated CI workflow to run snapshot tests
- Upgraded dependencies in uv.lock
- Fixed anchor link generation in session navigation

## Summary by CodeRabbit

* **New Features**
  * New --projects-dir CLI option to override default projects location
  * Added integration test target and HTML snapshot test suite

* **Improvements**
  * Parallel test execution enabled (pytest -n auto)
  * Project summary/index generation now prefers cached data with resilient fallback
  * More robust project discovery and CLI path handling
  * Session navigation anchors refined for expandable view

* **Documentation**
  * Updated README, test docs, and integration test data references

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: test/test_data/real_projects/-Users-dain-workspace-coderabbit-review-helper/agent-5e37fdd5.jsonl [hunk 1]
```diff
@@ -0,0 +1 @@
...:"external","cwd":"/Users/dain/workspace/coderabbit-review-helper","sessionId":"4227a68e-2b1f-4923-b824-6e04a67f77c1","version":"2.0.55","gitBranch":"uv-packaging","agentId":"5e37fdd5","message":{"model":"claude-haiku-4-5-20251001","id":"msg_01J9yyeVntmpSVKKiirSQwui","type":"message","role":"assistant","content":[{"type":"text","text":"I'm ready to help you explore and search through your codebase. I'm in read-only mode and will use my file search capabilities to help you find and analyze code efficiently.\n\nI can see you're working on the `uv-packaging` branch with recent work on packaging with `uv` and auto-detecting PRs from the current branch. Your project appears to be related to CodeRabbit feedback extraction.\n\nWhat would you like me to search for or explore in your codebase?"}],"stop_reason":null,"stop_sequence":null,"usage":{"input_tokens":1213,"cache_creation_input_tokens":0,"cache_read_input_tokens":0,"cache_creation":{"ephemeral_5m_input_tokens":0,"ephemeral_1h_input_tokens":0},"output_tokens":7,"service_tier":"standard"}},"requestId":"req_011CVYPou6i3uZxVLW12eWkm","type":"assistant","uuid":"84a68894-44b8-4596-8bc2-f7378c7a03e4","timestamp":"2025-11-27T12:28:22.113Z"}
```
[the remark this answers is not in the document]
Comment (daaain) [reply test/test_data/real_projects/-Users-dain-workspace-coderabbit-review-helper/agent-5e37fdd5.jsonl:1]: dw, these are open source projects

File: test/test_integration_realistic.py [hunk 2]
```diff
@@ -0,0 +1,1536 @@
...
+        # This tests that when we have a custom projects_dir, the matching still works
+        results = find_projects_by_cwd(
+            temp_projects_copy, "/Users/dain/workspace/JSSoundRecorder"
+        )
+
+        # The project should be found if path conversion works correctly
+        assert results, (
+            "Expected find_projects_by_cwd to return at least one matching project"
+        )
+        assert any("JSSoundRecorder" in r.name for r in results)
```
Comment (daaain) [inline test/test_integration_realistic.py:559]: It's a bit rubbish to skip this, would be good to test this on Windows too somehow
